src/backend/base/langflow/services/job_queue/service.py

```diff
@@ -0,0 +1,248 @@
+from __future__ import annotations
+
+import asyncio
+
+from loguru import logger
+
+from langflow.events.event_manager import EventManager, create_default_event_manager
+from langflow.services.base import Service
+
+
+class JobQueueService(Service):
+    """Asynchronous service for managing job-specific queues and their associated tasks.
+
+    This service allows clients to:
+      - Create dedicated asyncio queues for individual jobs.
+      - Associate each queue with an EventManager, enabling event-driven handling.
+      - Launch and manage asynchronous tasks that process these job queues.
+      - Safely clean up resources by cancelling active tasks and emptying queues.
+      - Automatically perform periodic cleanup of inactive or completed job queues.
+
+    Attributes:
+        name (str): Unique identifier for the service.
+        _queues (dict[str, tuple[asyncio.Queue, EventManager, asyncio.Task | None]]):
+            Dictionary mapping job IDs to a tuple containing:
+              * The job's asyncio.Queue instance.
+              * The associated EventManager instance.
+              * The asyncio.Task processing the job (if any).
+        _cleanup_task (asyncio.Task | None): Background task for periodic cleanup.
+        _closed (bool): Flag indicating whether the service is currently active.
+
+    Example:
+        service = JobQueueService()
+        await service.start()
+        queue, event_manager = service.create_queue("job123")
+        service.start_job("job123", some_async_coroutine())
+        # Retrieve and use the queue data as needed
+        data = service.get_queue_data("job123")
+        await service.cleanup_job("job123")
+        await service.stop()
+    """
+
+    name = "job_queue_service"
+
+    def __init__(self) -> None:
+        """Initialize the JobQueueService.
+
+        Sets up the internal registry for job queues, initializes the cleanup task, and sets the service state
+        to active.
+        """
+        self._queues: dict[str, tuple[asyncio.Queue, EventManager, asyncio.Task | None]] = {}
+        self._cleanup_task: asyncio.Task | None = None
+        self._closed = False
+
+    async def start(self) -> None:
```

Comment:
It seems it doesn't need to be async as it doesn't call `await` ?

Reply:
Fixed.